s3-spark/src/main/scala/geotrellis/spark/store/s3/SaveToS3.scala

```diff
@@ -74,7 +72,7 @@ object SaveToS3 {
     rdd.foreachPartition { partition =>
       val s3client = s3Client
       val requests: fs2.Stream[IO, (PutObjectRequest, RequestBody)] =
-        fs2.Stream.fromIterator[IO](
+        fs2.Stream.fromBlockingIterator[IO](
```

Comment:
By the `S3AsyncClient` usage we may eliminate a lot of `blocking` calls, simplify the code, and make it more resilient, see https://github.com/locationtech/geotrellis/issues/3467